File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression/Binary/Or.cs
using AST.Nodes.Interfaces;
using AST.Representation;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AST.Nodes.Expression.Binary
{
    public class Or : ASTNode, IExpression, IBinary
    {
        private readonly IExpression left;
        private readonly IExpression right;
        private string parsedString;

        public Or(IExpression left, IExpression right, string parsedString, PositionInText position)
         : base(position)
        {
            this.left = left;
            this.right = right;
            this.parsedString = parsedString;
        }
        public override string GetParsedString()
        {
            return parsedString;
        }

        public IExpression Left()
        { return left; }
        public IExpression Right()
        { return right; }

        public void Accept(Visitors.IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public T Accept<T>(Visitors.IVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public string MakeString()
        {
            return "||";
        }

        public Types.Type GetCompatibleType(Types.BoolType leftType, Types.BoolType rightType)
        {
            return new Types.BoolType();
        }

        public Types.Type GetCompatibleType(Types.Type leftType, Types.Type rightType)
        {
            return new Types.UndefinedType();
        }
    }
}

File: KevinRene/src/ql/ast/expression/booleanalgebra/Or.java
package ql.ast.expression.booleanalgebra;

import ql.ast.Expression;
import ql.ast.QLType;
import ql.ast.expression.Binary;
import ql.ast.type.QLBoolean;
import ql.ast.visitor.ExpressionVisitor;

public class Or extends Binary {
	public Or(Expression left, Expression right) {
		super(left, right, "||");
	}
	
	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
	@Override
	public QLType getType() {
		return new QLBoolean();
	}
}
